src/ossia-max/src/ZeroconfOscqueryListener.cpp

```diff
@@ -34,9 +33,56 @@ std::mutex ZeroconfOscqueryListener::m_mutex;
 
   void ZeroconfOscqueryListener::instanceAdded(const std::string& instance)
   {
-    std::lock_guard<std::mutex> lock(ZeroconfOscqueryListener::m_mutex);
+    std::lock_guard<std::mutex> lock(ZeroconfOscqueryListener::s_mutex);
+    s_connection_events.push_back({ZeroconfOscqueryListener::ConnectionEvent::ADDED, instance});
+  }
+
+  void ZeroconfOscqueryListener::instanceRemoved(const std::string& instance)
+  {
+    std::lock_guard<std::mutex> lock(ZeroconfOscqueryListener::s_mutex);
+    s_connection_events.push_back({ZeroconfOscqueryListener::ConnectionEvent::REMOVED, instance});
+  }
+
+  std::shared_ptr<ossia::net::generic_device> ZeroconfOscqueryListener::find_device(
+      const std::string& instance)
+  {
+    std::lock_guard<std::mutex> lock(ZeroconfOscqueryListener::s_mutex);
+
+    auto it = ossia::find_if(s_devices, [&](const auto& d) {
+      return d->get_name() == instance;
+    });
+
+    if (it != s_devices.end())
+    {
+      return *it;
+    }
+    return nullptr;
+  }
 
-    for (const auto& dev : m_devices)
+  void ZeroconfOscqueryListener::browse()
+  {
+    s_mutex.lock();
```

Comment:
I'd go for std::lock_guard here as this could cause the mutex to always be locked if something underneath throws

Reply:
done in ceb3a6b7